pandas/core/arrays/interval.py

```diff
@@ -562,6 +564,117 @@ def from_tuples(
+            When the dtype of the string cannot be parsed as either float,
+            Timestamp or Timedelta
+
+        See Also
+        --------
+        interval_range : Function to create a fixed frequency IntervalIndex.
+        %(klass)s.from_breaks : Construct an %(klass)s from an array of
+            splits.
+        %(klass)s.from_tuples : Construct an %(klass)s from an
+            array-like of tuples.
+
+        %(examples)s\
+        """
+    )
+
+    @classmethod
+    @Appender(
+        _interval_shared_docs["from_strings"]
+        % {
+            "klass": "IntervalIndex",
+            "examples": textwrap.dedent(
+                """\
+        Examples
+        --------
+        >>> pd.IntervalIndex.from_strings(["(0, 1]", "(1, 2]"])
+        IntervalIndex([(0, 1], (1, 2]],
+                       dtype='interval[int64, right]')
+        """
+            ),
+        }
+    )
+    def from_strings(
+        cls: type[IntervalArrayT],
+        data: Sequence[str],
+        closed: str = "right",
+        dtype: Dtype | None = None,
+    ) -> IntervalArrayT:
+        # These need to be imported here to avoid circular dependencies.
+        from pandas import (
+            to_datetime,
+            to_timedelta,
+        )
+
+        pattern = re.compile(r"\(.*,.*]")
+
+        left, right = [], []
+        for string in data:
+
+            # Try to match "(left, right]" where 'left' and 'right' are breaks.
+            breaks_match = pattern.match(string)
+
+            if breaks_match is None:
+                raise ValueError(
+                    "Could not find opening '(' and closing ']' "
+                    f"brackets in string: '{string}'"
+                )
+            # Try to split 'left' and 'right' based on a comma and a space.
+            breaks = breaks_match.string[1:-1].split(", ", 1)
+
+            if len(breaks) != 2:
+                raise ValueError(
+                    f"Delimiter ', ' (comma + space) not found in string: {string}"
+                )
+
+            conversions: list[Callable] = [int, float, to_datetime, to_timedelta]
+            # Try to parse the breaks first as floats, then datetime, then timedelta.
+            for i, conversion in enumerate(conversions):
+                # Check if all breaks can be parsed as integers.
+                if i == 0 and not all(b.isdigit() for b in breaks):
+                    continue
+                try:
+                    newleft, newright = map(conversion, breaks)
+                    left.append(newleft)
+                    right.append(newright)
+                    break
+                except ValueError:
+                    continue
+            else:
+                raise ValueError(
+                    "Could not parse string as Interval of float, Timedelta "
+                    f"or Timestamp: {string}"
+                )
+
+        # If dtype was not an IntervalDtype, try to parse it as such.
```

Comment:
can you document that dtype is a *scalar* type, so for example IntervalDtype should actually not be allowed here.

Reply:
pls add tests for this

Reply:
I'm sorry but I'm afraid I don't understand your comment here. 

`IntervalIndex.from_arrays()` requires an `IntervalDtype` upon instantiation. If one is given as the `dtype` argument to `IntervalIndex.from_strings()`, that works fine. If not, the `dtype` will be converted appropriately.